lib/os/dec.c

```diff
@@ -25,7 +25,7 @@ uint8_t u8_to_dec(char *buf, uint8_t buflen, uint8_t value)
 		divisor /= 10;
 	}
 
-	if (buflen) {
+	if (buflen != 0) {
```

Comment:
Result of not-equal-to operator has type int, this change does not make any sense.